### PR TITLE
HOSTEDCP-1328: Add Ability to Encrypt VM OS Disks

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
Adds the ability to encrypt the OS disks in the Azure VMs, using a DiskEncryptionSet resource. The DiskEncryptionSet ID can be passed in when creating a new Hosted Cluster or creating a new NodePool.

> **Note:** this PR is also enabling encryption at host in addition to the server side encryption through the DiskEncryptionSet. See this [link](https://learn.microsoft.com/en-us/azure/virtual-machines/disk-encryption-overview) for more details.

**Which issue(s) this PR fixes**:
Fixes [HOSTEDCP-1328](https://issues.redhat.com/browse/HOSTEDCP-1328)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [x] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: cmd/cluster/azure/create.go [hunk 1]
```diff
@@ -30,35 +28,37 @@ func NewCreateCommand(opts *core.CreateOptions) *cobra.Command {
...
+	cmd.Flags().StringSliceVar(&opts.AzurePlatform.AvailabilityZones, "availability-zones", opts.AzurePlatform.AvailabilityZones, "The availability zones in which NodePools will be created. Must be left unspecified if the region does not support AZs. If set, one nodepool per zone will be created.")
 	cmd.Flags().StringVar(&opts.AzurePlatform.ResourceGroupName, "resource-group-name", opts.AzurePlatform.ResourceGroupName, "A resource group name to create the HostedCluster infrastructure resources under.")
+	cmd.Flags().StringVar(&opts.AzurePlatform.DiskEncryptionSetID, "disk-encryption-set-id", opts.AzurePlatform.DiskEncryptionSetID, "The Disk Encryption Set ID to use to encrypt the OS disks for the VMs.")
 
 	_ = cmd.MarkFlagRequired("azure-creds")
 	_ = cmd.MarkPersistentFlagRequired("pull-secret")
 
-	cmd.Run = func(cmd *cobra.Command, args []string) {
-		ctx, cancel := context.WithCancel(context.Background())
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
```
Comment (csrwng) [inline cmd/cluster/azure/create.go:44]: Is there a reason you're changing to RunE here? If I remember correctly, when you use RunE and an error is thrown, the default behavior is to print the error and then the usage, which is confusing. You may want to try throwing an error and see whether that's the behavior you want.

**Review by csrwng — COMMENTED** [review]
Just one comment, otherwise lgtm

File: cmd/cluster/azure/create.go (hunk 1 above)
Comment (bryan-cox) [reply cmd/cluster/azure/create.go:44]: I was updating it to be consistent with how this same command structure is run in AWS, Agent, and Kubevirt.

**Comment by bryan-cox** [comment]
Just for reference - I added upstream documentation as part of the work for this PR [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4415).

**Pull request merged**